src/release.rs

```diff
@@ -145,4 +172,37 @@ impl BuildConfig {
         info!("Build took {:.3?} seconds", start_time.elapsed());
         Ok(())
     }
+
+    async fn download_release_from_channel(&self, file_name: &str) -> Result<(), Box<dyn Error>> {
+        info!("Downloading release from channel: {}", self.release_channel);
+        let tar_file = format!("{}{}", file_name, ".tar.bz2");
```

Comment:
nit: Rather than reconstructing the tar filename, how about passing it in directly as `tar_filename` instead of `file_name`?